SwEnt-Group8/Street-work-app
# Feat/friend screen in profile screen

## Friend screen in profile

![Profile-with-basic-friend-screen](https://github.com/user-attachments/assets/fa9d75f4-be6e-4785-a0dc-5b3d88664e41)

![Profile-with-basic-friend-screen-empty](https://github.com/user-attachments/assets/70abcc59-7529-40d3-9428-b97b8e519e0e)

### Profile UI

- Now displays a **list of friends** (hardcoded list to cirumvent MVVM issues).
- If **no friends** (empty list), then the display is replaced with a **text**.
- Also now display current user (placeholder) **score** and **username**.
- Removed **QR-code** placeholder because we will use **NFC** or **UID-adding**.
- Clicking on the **train with a friend** button will now display a toast saying that it is not implemented yet.

### Components to display friends

- `DisplayFriend()` to display one element in the friend list
- `DisplayFriendList()` to display a list of friends (User).

Currently displayed friend information is :
- The username
- The score
- A placeholder profile picture

### Tests

- `ProfileScreenTest()` has been **updated**.
- `ProfileFriendListTest()` verifies that the components to display friends are working properly.

### To test the feature

Simply run the `mainActivity` and go to the **profile screen**.
You can uncomment the following line : `friendList = emptyList()` in order to verify the no friends behavior

### Improvements for future tasks

- Make the friend screen UI looks like the Figma.
- MVVM linking works (fetching currentUser and friends).
